### ISSUE TITLE
[Feature request] Remove write access in repos to stop students from pushing more codes

### ISSUE THREAD

**Issue by PeterDaveHello** [education/teachers_pet#112]
I think assignment/homework usually has a deadline, thought we could checkout to commit before a datetime, but the commit log is easily to be faked, also sometimes student didn't configure the datetime and timezone on there machine, so I think so best way to make sure the code will be committed before deadline is to remove the write permission, which is not supported by teachers_pet currently, hope it can be added in, thanks!

**Comment by mikehelmick** [comment]
FYI - when I originally wrote this I would clone all the repositories to my local machine and then roll them back to the due date.

This allows you to have individual extensions for individual students easily.

**Comment by PeterDaveHello** [comment]
Yeah, I can easily fetch the remotes at time deadline, but I just wonder if we could have this feature, 'cause we(teacher/ta/student) could make sure we have a common view on GitHub to make sure the status of assignment.
